src/perform_step/bdf_perform_step.jl

```diff
@@ -733,24 +733,35 @@ function perform_step!(integrator,cache::QNDFConstantCache,repeat_step=false)
       γₖ[i] = 1//1
     end
   end
-  γ = inv((1-κ)*γₖ[k])
-  nlsolver.γ = γ
+  β₀ = inv((1-κ)*γₖ[k])
+  α₀ = 1
+
   # precalculations
   u₀ = uprev + sum(D)  # u₀ is predicted value
   ϕ = zero(u)
   for i = 1:k
-    ϕ += γₖ[i]*D[i] # Done .+ because of +(::Rational{Int64}, ::Array{Float64,1}) use case
+    ϕ += γₖ[i]*D[i]
   end
-  ϕ *= γ
-  nlsolver.tmp = u₀ - ϕ
-  γdt = γ*dt
+  ϕ *= β₀
   markfirststage!(nlsolver)
   # initial guess
-  nlsolver.z = (uprev + sum(D) -nlsolver.tmp)*inv(γ)
-  z = nlsolve!(nlsolver, integrator, cache, repeat_step)
-  nlsolvefail(nlsolver) && return
-  u = nlsolver.tmp + γ*z
+  nlsolver.z = uprev + sum(D)
+
+  mass_matrix = f.mass_matrix
+
+  if mass_matrix == I
+    nlsolver.tmp = @.. (u₀ - ϕ)/ (dt * β₀)
+  else
+    _tmp = mass_matrix * @.. (u₀ - ϕ)
+    nlsolver.tmp = @.. _tmp / (dt * β₀)
+  end
+  nlsolver.γ = β₀
+  nlsolver.α = α₀
+  nlsolver.method = COEFFICIENT_MULTISTEP
```

Comment:
I'm a bit surprised about the mix of broadcasted and non-broadcasted expressions (not only in these lines, here it would be just `uprev .+ sum(D)`, but also above e.g. in `ϕ += γₖ[i] * D[i]`). Shouldn't we consistently use broadcasting? Or does it still impact performance?

Reply:
Yeah I agree, actually, there's some dimensional problem like that D is 1x5 matrix and γ is 6-Array, it was originally defined like that. we can change it probably for more consistent operation using broadcast, maybe in another PR.